src/phalanx/cli.py

```diff
@@ -334,6 +353,93 @@ def environment() -> None:
     """Commands for Phalanx environment configuration."""
 
 
+@environment.command("install")
+@click.argument("environment")
+@click.option(
+    "-c",
+    "--config",
+    type=click.Path(path_type=Path),
+    default=None,
+    help="Path to root of Phalanx configuration.",
+)
+@click.option(
+    "--git-branch",
+    default=None,
+    envvar="GITHUB_HEAD_REF",
+    help="Override Git branch for Argo CD.",
+)
+@click.option(
+    "--force-noninteractive",
+    default=False,
+    is_flag=True,
+    help="Force installation without a prompt.",
+)
+@click.option(
+    "--vault-role-id",
+    default=None,
+    envvar="VAULT_ROLE_ID",
+    help="Role ID for vault-secrets-operator.",
+)
+@click.option(
+    "--vault-secret-id",
+    default=None,
+    envvar="VAULT_SECRET_ID",
+    help="Secret ID for vault-secrets-operator.",
+)
+@click.option(
+    "--vault-token",
+    default=None,
+    envvar="VAULT_TOKEN",
+    help="Read-only token for vault-secrets-operator.",
+)
+def environment_install(
+    environment: str,
+    *,
+    config: Path | None,
+    force_noninteractive: bool = False,
+    git_branch: str | None = None,
+    vault_role_id: str | None = None,
+    vault_secret_id: str | None = None,
+    vault_token: str | None = None,
+) -> None:
+    """Install Phalanx into an environment.
+
+    Bootstrap Phalanx for an environment. Assumes that the currently enabled
+    Kubernetes configuration is the cluster into which to install Phalanx.
+
+    The secrets tree for the environment must already be present in Vault.
+    Read-only Vault credentials must be supplied by either setting the
+    environment variables VAULT_ROLE_ID and VAULT_SECRET_ID to the credentials
+    of a Vault AppRole, or setting VAULT_TOKEN to a read-only Vault token.
+    """
+    if not config:
+        config = _find_config()
+    factory = Factory(config)
+    if vault_role_id and vault_secret_id:
+        vault_credentials: VaultCredentials = VaultAppRoleCredentials(
+            role_id=vault_role_id, secret_id=vault_secret_id
+        )
+    elif vault_token:
+        vault_credentials = VaultTokenCredentials(token=vault_token)
+    else:
+        msg = (
+            "Either VAULT_TOKEN or both VAULT_ROLE_ID and VAULT_SECRET_ID"
+            " must be set"
+        )
+        raise click.UsageError(msg)
+
+    # Prompt the user unless they specifically said not to.
+    if not force_noninteractive:
+        print(_INSTALL_WARNING.format(environment=environment))
```

Comment:
I don't know if click.confirm will let you do this, and maybe it's being TOO aggressive...

...but given my tendency to speed through installers without thinking about them, I feel like it might be a good idea to generate a nonce -- maybe the classical six-digit random PIN -- display it, and make the user type it or at least copy-paste it, rather than just "y-Return".

Reply:
I'm going to merge this in its current form, but I added to my to-do list determining the current configured Kubernetes cluster and displaying that as part of the prompt to make it easier for people to realize when they're pointing to the wrong cluster.